x-pack/plugin/vector-tile/src/test/java/org/elasticsearch/xpack/vectortile/feature/FeatureFactoryTests.java

```diff
@@ -292,6 +293,44 @@ private void assertParsing(InputStream is) throws IOException, ParseException {
         assertThat(builder.getFeatures(geometry), iterableWithSize(1));
     }
 
+    public void testVectorTilesDoNotContainOriginWithSimplifiableGeometries() throws IOException, ParseException {
+        Polygon polygon = new Polygon(
+            new LinearRing(
+                new double[] { -94.491034, -94.493892, -94.399587, -94.483604, -94.491034 },
+                new double[] { -72.522177, -72.491761, -72.607412, -72.545519, -72.522177 }
+            )
+        );
+
+        final FeatureFactory builder = new FeatureFactory(0, 0, 0, 4096, 5);
+
+        var mvtGeometry = builder.getMvtGeometry(polygon);
+        var coordinates = mvtGeometry.getCoordinates();
+        var containsCenter = Arrays.stream(coordinates).anyMatch(c -> c.x == 2048 && c.y == 2048);
+        assertFalse(containsCenter);
```

Comment:
We could assert here on the exact contents of the vector tiles, but they are jts version dependent, and it changes from 1.15 so 1.20, so I tried to add a version independent check instead.